articles/service-fabric/service-fabric-reliable-services-communication-remoting.md

```diff
@@ -219,90 +219,80 @@ Following example uses Json Serialization with Remoting V2.
         public IServiceRemotingRequestMessageBodySerializer CreateRequestMessageSerializer(Type serviceInterfaceType,
             IEnumerable<Type> requestBodyTypes)
         {
-            return new ServiceRemotingRequestJsonMessageBodySerializer(serviceInterfaceType, requestBodyTypes);
+            return new ServiceRemotingRequestJsonMessageBodySerializer();
         }
-
+    
         public IServiceRemotingResponseMessageBodySerializer CreateResponseMessageSerializer(Type serviceInterfaceType,
             IEnumerable<Type> responseBodyTypes)
         {
-            return new ServiceRemotingResponseJsonMessageBodySerializer(serviceInterfaceType, responseBodyTypes);
+            return new ServiceRemotingResponseJsonMessageBodySerializer();
         }
-
+    
         public IServiceRemotingMessageBodyFactory CreateMessageBodyFactory()
         {
             return new JsonMessageFactory();
         }
     }
-
+    
     class JsonMessageFactory : IServiceRemotingMessageBodyFactory
     {
         public IServiceRemotingRequestMessageBody CreateRequest(string interfaceName, string methodName,
             int numberOfParameters)
         {
             return new JsonRemotingRequestBody();
         }
-
+    
         public IServiceRemotingResponseMessageBody CreateResponse(string interfaceName, string methodName)
         {
             return new JsonRemotingResponseBody();
         }
     }
-
+    
     class ServiceRemotingRequestJsonMessageBodySerializer : IServiceRemotingRequestMessageBodySerializer
     {
-        public ServiceRemotingRequestJsonMessageBodySerializer(Type serviceInterfaceType,
-            IEnumerable<Type> parameterInfo)
-        {
-        }
-
         public OutgoingMessageBody Serialize(IServiceRemotingRequestMessageBody serviceRemotingRequestMessageBody)
         {
             if (serviceRemotingRequestMessageBody == null)
             {
                 return null;
             }
-
-            var writeStream = new MemoryStream();
-            var jsonWriter = new JsonTextWriter(new StreamWriter(writeStream));
-
-            var serializer = JsonSerializer.Create(new JsonSerializerSettings()
+    
+            using (var writeStream = new MemoryStream())
+            using (var jsonWriter = new JsonTextWriter(new StreamWriter(writeStream)))
             {
-                TypeNameHandling = TypeNameHandling.All
-            });
-            serializer.Serialize(jsonWriter, serviceRemotingRequestMessageBody);
-
-            jsonWriter.Flush();
-            var segment = new ArraySegment<byte>(writeStream.ToArray());
-            var segments = new List<ArraySegment<byte>> { segment };
-            return new OutgoingMessageBody(segments);
+                var serializer = JsonSerializer.Create(new JsonSerializerSettings()
+                {
+                    TypeNameHandling = TypeNameHandling.All
+                });
+                serializer.Serialize(jsonWriter, serviceRemotingRequestMessageBody);
```

Comment:
@brutaldev 
This is first time you are serializing the serviceRemotingRequestMessageBody.